### ISSUE TITLE
Multiple connections use single CopyingByteBufferAllocator instances?

### ISSUE THREAD

**Issue by beetlex-io** [txdv/LibuvSharp#18]
UVStream 

 ByteBufferAllocatorBase allocator;
        public ByteBufferAllocatorBase ByteBufferAllocator
        {
            get
            {
                return allocator ?? Loop.ByteBufferAllocator;
            }
            set
            {
                allocator = value;
            }
        }
  public void Resume()
        {
            CheckDisposed();
            int r = uv_read_start(NativeHandle, ByteBufferAllocator.AllocCallback, read_cb);
            Ensure.Success(r);
        }

**Comment by txdv** [comment]
Yeah they do, you can assign your own bytebufferallocator, for every handler and loop.

This one just creates a new buffer and copies the contents to it, so everyone can use it.
